bosh_aws_cpi/lib/cloud/aws/stemcell_creator.rb

```diff
@@ -22,7 +22,7 @@ def create(volume, ebs_volume, image_path)
       ResourceWait.for_snapshot(snapshot: snapshot, state: :completed)
 
       params = image_params(snapshot.id)
-      image = region.images.create(params)
+      image = region.images[region.client.register_image(params).image_id]
```

Comment:
You may have to create a mock client off of region which has register_image method stubbed out. It should return image_id and is one of the images in the region.images.